client/src/components/Login.js

```diff
@@ -24,11 +24,15 @@ const ApplicantLogin = () => {
         const token = data.token;
         localStorage.setItem("token", token);
         localStorage.setItem("user", JSON.stringify(data.user));
-        setSuccess(true);
+        if (data.user.role === "applicant") {
```

Comment:
You need to check first if there is a user return as part of data then access a user role, so if a user was undefined the app will crach.
E.g ` if (data.user && data.user.role === "applicant")`

Reply:
Done. Thanks.